src/hotspot/share/runtime/objectMonitor.inline.hpp

```diff
@@ -150,6 +150,11 @@ inline void ObjectMonitor::set_recursions(size_t recursions) {
   _recursions = checked_cast<intx>(recursions);
 }
 
+inline void ObjectMonitor::increment_recursions(JavaThread* current) {
+  assert(has_owner(current), "must be the owner");
```

Comment:
May be overkill to have this assert ...

Reply:
Yeah this method will be removed with legacy locking.